### DISCUSSION TITLE
XIAO nRF52840 Sense sleep problem

### DISCUSSION THREAD

**Discussion by MrTinkerman** [micropython/micropython#14024] [nRF] [answered]
Hi,

I have a XIAO nRF52840 Sense and I'm trying to use lightsleep.

An RFM95W Lora board is connected to it which is put in sleep mode when not transmitting.
I'm also using the onboard IMU of the XIAO.

To save additional power I tried putting the XIAO into sleep mode.
Results for different firmware versions are below. (https://micropython.org/download/SEEED_XIAO_NRF52/)

Am I missing something or is lightsleep broken?

(Power consumption measured using a Nordic Power Profiler Kit on the battery connectors of the XIAO )
```
SEEED_XIAO_NRF52-20230426-v1.20.0
time.sleep_ms(int(1000/samplerate))  2.35 mA
lightsleep(int(1000/samplerate))     TypeError: function takes 0 positional arguments but 1 were given

SEEED_XIAO_NRF52-20231005-v1.21.0
time.sleep_ms(int(1000/samplerate))  2.37 mA
lightsleep(int(1000/samplerate))     TypeError: function takes 0 positional arguments but 1 were given

SEEED_XIAO_NRF52-20231227-v1.22.0
time.sleep_ms(int(1000/samplerate))  2.40 mA
lightsleep(int(1000/samplerate))     8.42 mA

SEEED_XIAO_NRF52-20240105-v1.22.1
time.sleep_ms(int(1000/samplerate))  2.39 mA
lightsleep(int(1000/samplerate))     8.42 mA

SEEED_XIAO_NRF52-20240304-v1.23.0-preview.176.g90e517862
time.sleep_ms(int(1000/samplerate))  2.42 mA
lightsleep(int(1000/samplerate))     8.40 mA
```

Edit: Fixed typo

**Comment by robert-hh** [comment, marked as answer]
lightsleep and deepsleep are not implemented for the NRF port. The methods exist, but do almost nothing.
lightsleep() just calls __WFE(), deepsleep() just resets the boards.
Edit: But the XIAO board should behave the same as the other one. Maybe update the firmware. lightsleep() and deepsleep() were moved to extmod in Nov 2023.

**Reply by MrTinkerman** [reply]
Thanks for mentioning, 2.5 mA is acceptable though so I'll just omit the sleep then.

I guess "the other one" you are referring to is the Nordic PPK2: 
https://www.nordicsemi.com/Products/Development-hardware/Power-Profiler-Kit-2
